### PR TITLE
#0: Add CMake build flag ENABLE_LIBCXX to selectively enable/disable libc++

### PR DESCRIPTION
**Pull request by nsmithtt**
This PR enables users of ttnn/ttmetal to disable libc++ if they want to link against libstdc++ instead.

### CODE REVIEW

File: tt_metal/tt_stl/aligned_allocator.hpp [hunk 1]
```diff
@@ -43,7 +43,7 @@ class aligned_allocator {
         return static_cast<T*>(p);
     }
 
-    void deallocate(T* p, std::size_t size) const noexcept { ::operator delete[](p, size * sizeof(T), alignment); }
+    void deallocate(T* p, std::size_t size) const noexcept { ::operator delete[](p, alignment); }
```
Comment (TT-billteng) [inline tt_metal/tt_stl/aligned_allocator.hpp:46]: why is this change needed?

Comment (nsmithtt) [reply tt_metal/tt_stl/aligned_allocator.hpp:46]: It wasn't present in libstdc++'s header, but after a bit more digging it seems like it can be enabled if we compile with flag ` -fsized-deallocation`.  In reading cppref I can't tell that there's any difference between APIs 3-4 and 7-8. https://en.cppreference.com/w/cpp/memory/new/operator_delete

Comment (nsmithtt) [reply tt_metal/tt_stl/aligned_allocator.hpp:46]: I can update the PR to use `-fsized-deallocation` if that seems better.

File: CMakeLists.txt [hunk 2]
```diff
@@ -72,6 +72,8 @@ option(ENABLE_CODE_TIMERS "Enable code timers" OFF)
 
 option(ENABLE_TRACY "Enable Tracy Profiling" OFF)
 
+option(ENABLE_LIBCXX "Disable using libc++" ON)
```
Comment (marty1885) [inline CMakeLists.txt:75]: This comment is confusing. The variable name indicates setting this variable to ON means to enable lib++ but comment says **disable**

Comment (nsmithtt) [reply CMakeLists.txt]: Thanks! I renamed the variable and not the comment :)

**Review by TT-billteng — APPROVED** [review]

**Review by tt-rkim — APPROVED** [review]
